src/hooks/useLocalStorage.ts

```diff
@@ -60,6 +65,13 @@ export const useLocalStorage = <T>({
     (value) => {
       storageSet(key, value);
       setState(value);
+      if (expire) {
+        const expiryDate = DateTime.now().plus({ seconds: expire });
+        const expiryKey = `${key}${storageKeys.EXPIRY_DATE}`;
+        storageSet(expiryKey, expiryDate);
+        debounce(storageService.remove, expire * 1000)(key);
```

Comment:
While not added as part of this PR, I think the `debounce` function is confusing. Because if you compare it to the [debounce](https://docs-lodash.com/v4/debounce/) function of lodash, they are quite different actually. And since the function is actually just a more convenient implementation to create a timer, I would rename it to indicate such. Maybe rename it to `createTimer()` or `periodicCall()` or whatever.

Reply:
Once this conversation is resolved, please fell free to merge :)

Reply:
So debounce is debouncing the writing to the storage. This is used cause we write data to localstorage on input change which can be quite frequent so I am debouncing this for 1s or so.

Reply:
Ahhhh now I get it. I missed the point of the `let timer ...` outside of the timeout function. Okay, then I guess that works.